numerical_methods/secant_method.c

```diff
@@ -0,0 +1,84 @@
+/**
+ * @file
+ * @brief [Wikipedia link](https://en.wikipedia.org/wiki/Secant_method). Find a
+ * continuous function's root by using a succession of roots of secant lines to
+ * approximate it, starting from the given points' secant line.
+ * @author [Samuel Pires](https://github.com/disnoca)
+ */
+
+#include <assert.h>  /// for assert
+#include <math.h>    /// for fabs
+#include <stdio.h>   /// for io operations
+
+#define TOLERANCE 0.0001  // root approximation result tolerance
+#define NMAX 100          // maximum number of iterations
+
+/**
+ * @brief Continuous function for which we want to find the root
+ * @param x Real input variable
+ * @returns The evaluation result of the function using the input value
+ */
+double func(double x)
```

Comment:
The name seems a bit ambiguous i think it might be better to have a more descriptive name

Reply:
What kind of name do you suggest? This is just the function that will be used in the algorithm, it can't get any more descriptive than this. Most other numerical methods algorithms use this same name or just f.

Reply:
I see